client/app/src/main/java/com/microsoft/research/karya/ui/accesscode/AccessCodeFragment.kt

```diff
@@ -0,0 +1,137 @@
+import com.microsoft.research.karya.utils.Result
+import com.microsoft.research.karya.utils.SeparatorTextWatcher
+import com.microsoft.research.karya.utils.extensions.dataStore
+import com.microsoft.research.karya.utils.extensions.gone
+import com.microsoft.research.karya.utils.extensions.observe
+import com.microsoft.research.karya.utils.extensions.requestSoftKeyFocus
+import com.microsoft.research.karya.utils.extensions.visible
+import com.microsoft.research.karya.utils.viewBinding
+import kotlinx.coroutines.launch
+
+class AccessCodeFragment : Fragment(R.layout.fragment_access_code) {
+    private val binding by viewBinding(FragmentAccessCodeBinding::bind)
+    private val viewModel by activityViewModels<AccessCodeViewModel>()
+
+    private val creationCodeLength = 16
+    private val creationCodeEtMax = creationCodeLength + (creationCodeLength - 1) / 4
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+        observe()
+    }
+
+    private fun observe() {
+        viewModel.stateFlow.observe(lifecycle, lifecycleScope) {
+            println(it)
+        }
+    }
+
+    private fun setupViews() {
+        with(binding) {
+            /** Add text change listener to creation code */
+            creationCodeEt.addTextChangedListener(object : SeparatorTextWatcher('-', 4) {
+                override fun onAfterTextChanged(text: String, position: Int) {
+                    creationCodeEt.run {
+                        setText(text)
+                        setSelection(position)
+                    }
+
+                    /** If creation code length has reached max, call handler */
+                    if (creationCodeEt.length() == creationCodeEtMax) {
+                        // TODO: call this once the user presses the button to move forward.
+                        handleFullCreationCode()
+                    } else {
+                        clearErrorMessages()
+                    }
+                }
+            })
+            requestSoftKeyFocus(creationCodeEt)
+        }
+    }
+
+    private fun checkAccessCode(accessCode: String) {
+        viewModel.checkAccessCode(accessCode).observe(lifecycle, lifecycleScope) { result ->
+            when (result) {
+                is Result.Success<*> -> onAccessCodeVerified(result as Result.Success<Int>, accessCode)
+                // TODO: Use error codes and exceptions from Anurag's PR
+                is Result.Error -> showErrorMessage(result.exception.message ?: "Error fetching data")
+                Result.Loading -> showLoading()
+            }
+        }
+    }
+
+    private fun onAccessCodeVerified(successResult: Result.Success<Int>, accessCode: String) {
+        showSuccessMessage()
+        lifecycleScope.launch { updateLanguagePreference(successResult.value) }
+        WorkerInformation.creation_code = accessCode
```

Comment:
Are we moving forward with WorkerInformation?

Reply:
Nope, but since you are using it during this sprint, I updated it so that your registration workflow can complete successfully.